documentation/release_notes.rst

```diff
@@ -47,6 +47,14 @@ New in This Release
 - ``sort``, ``stable_sort``, ``sort_by_key`` and ``stable_sort_by_key`` algorithms fail to compile
   when using Clang 17 and earlier versions, as well as compilers based on these versions,
   such as Intel(R) oneAPI DPC++/C++ Compiler 2023.2.0.
+- When compiling several algorithms with device execution policies, both the host and device compilation passes must be
```

Comment:
Should this be more general to point out the possible issue with the clang++ driver, or is it important to list the algorithms where we have a workaround in place to expose this issue? I would suggest the following general wording:

> When compiling code that uses device policies with the open source oneAPI DPC++ Compiler (clang++ driver) synchronous SYCL runtime exceptions regarding unfound kernels may be encountered unless and optimization flag is specified -- e.g., ``-O1`` -- instead of relying on the compiler's default optimization level.

Reply:
I adopted a more general wording with a few small changes from your suggestion.